src/Views/Text.vala

```diff
@@ -70,52 +70,51 @@ public class PantheonShell.Text : Gtk.Grid {
         margin_start = margin_end = 12;
         margin_bottom = 24;
         attach (text_size_label, 0, 0);
-        attach (text_size_modebutton, 1, 0, 2);
+        attach (text_size_scale, 1, 0, 2);
         attach (dyslexia_font_label, 0, 1);
         attach (dyslexia_font_switch, 1, 1);
         attach (dyslexia_font_description_label, 1, 2, 2);
 
         var interface_settings = new Settings ("org.gnome.desktop.interface");
-        var interface_font = interface_settings.get_string (FONT_KEY);
-        var document_font = interface_settings.get_string (DOCUMENT_FONT_KEY);
-        var monospace_font = interface_settings.get_string (MONOSPACE_FONT_KEY);
-
-        text_size_modebutton.set_active (get_text_scale (interface_settings));
-
-        interface_settings.changed.connect (() => {
-            text_size_modebutton.set_active (get_text_scale (interface_settings));
-        });
+        interface_settings.bind ("text-scaling-factor", text_size_adjustment, "value", SettingsBindFlags.GET);
 
-        text_size_modebutton.mode_changed.connect (() => {
-            interface_settings.set_double (TEXT_SIZE_KEY, TEXT_SCALE[text_size_modebutton.selected]);
+        // Setting scale is slow, so we wait to keep UI responsive
+        text_size_scale.button_release_event.connect (() => {
```

Comment:
So the problem with this approach is that it's not accessible with the keyboard. Gotta keep that. :smile: